### PR TITLE
br: restore checkpoint

### PR DESCRIPTION
**Pull request by Leavrth**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

br v7.1.0 supports restoring checkpoint.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [x] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/br-checkpoint-restore.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+## 使用限制
+
+断点恢复功能依赖于 GC 机制，且无法记录所有已恢复的数据，具体描述如下。
+
+### GC 会被停止
+
+在日志恢复过程中，日志数据恢复的顺序是无序的，因此可能会出现一个 key 的删除记录优先于该 key 的写入记录恢复的情况。如果此时触发 GC，该 key 的所有数据会被删除，导致该 key 后续恢复的写入记录无法被 GC 处理。为了避免这种情况，br 命令行工具会在日志恢复过程中暂停 GC。当 br 工具中途退出后，GC 将仍然保持暂停状态。
+
+日志恢复完成后会重新启动 gc，不需要手动启动。但如果你不想再继续恢复时，请手动开启 gc，开启方法如下：
+br 工具暂停 GC 的原理是执行 `SET config tikv gc.ratio-threshold = -1.0`，手动开启执行相似的 SQL，例如重置为默认值，你可以执行 SQL `SET config tikv gc.ratio-threshold = 1.1`
```
Comment (Oreoxmt) [inline br/br-checkpoint-restore.md:25]:
```suggestion
日志恢复完成后会重新启动 GC，不需要手动启动。但如果你不想再继续恢复时，请手动开启 GC，开启方法如下：

br 工具暂停 GC 的原理是执行 `SET config tikv gc.ratio-threshold = -1.0`，你可以通过修改 `gc.ratio-threshold` 的值手动开启 GC，例如执行 `SET config tikv gc.ratio-threshold = 1.1` 将参数重置为默认值。
```

File: br/br-checkpoint-restore.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+快照恢复的实现原理与[快照备份](/br/br-checkpoint-backup.md#实现原理)类似，br 工具会批量恢复一段键范围 (Region) 内的所有的 SST 文件。恢复完成后，br 工具记录下这段范围以及该范围对应的恢复集群的表 ID。断点恢复功能会定期将这些新增的相关恢复信息上传至外部存储中，从而持久化存储该恢复任务已完成的键范围。
+
+在重试恢复时，br 工具会从外部存储读取已经恢复的键范围，并匹配相关的表 ID。在恢复过程中，会跳过与断点恢复记录的键范围重叠且对应表 ID 相同的范围。
+
+如果在重试恢复前，你删除了某些表，那么重试恢复时新创建的表的 ID 会与之前记录在断点恢复的表 ID 不同，从而绕过使用之前的断点恢复信息，重新恢复该表。
+
+由于数据采用了 MVCC（多版本并发控制）机制，带有固定 TS 的数据可以被无序且重复写入。
+
+快照恢复在恢复库表 ddl 时添加了 `ifExists` 参数。对于已经存在的库表（视作已经创建完毕），br 工具会自动跳过恢复。
```
Comment (Oreoxmt) [inline br/br-checkpoint-restore.md:54]:
```suggestion
快照恢复在恢复库表 DDL 时添加了 `ifExists` 参数。对于已经存在的库表（视作已经创建完毕），br 工具会自动跳过恢复。
```

**Review by YuJuncen — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

File: br/br-checkpoint-restore.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+快照恢复的实现原理与[快照备份](/br/br-checkpoint-backup.md#实现原理)类似，br 工具会批量恢复一段键范围 (Region) 内的所有的 SST 文件。恢复完成后，br 工具记录下这段范围以及该范围对应的恢复集群的表 ID。断点恢复功能会定期将这些新增的相关恢复信息上传至外部存储中，从而持久化存储该恢复任务已完成的键范围。
+
+在重试恢复时，br 工具会从外部存储读取已经恢复的键范围，并匹配相关的表 ID。在恢复过程中，会跳过与断点恢复记录的键范围重叠且对应表 ID 相同的范围。
+
+如果在重试恢复前，你删除了某些表，那么重试恢复时新创建的表的 ID 会与之前记录在断点恢复的表 ID 不同，从而绕过使用之前的断点恢复信息，重新恢复该表。
+
+由于数据采用了 MVCC（多版本并发控制）机制，带有固定 TS 的数据可以被无序且重复写入。
+
+快照恢复在恢复库表 DDL 时添加了 `ifExists` 参数。对于已经存在的库表（视作已经创建完毕），br 工具会自动跳过恢复。
```
Comment (hfxsd) [inline br/br-checkpoint-restore.md:55]:
```suggestion
快照恢复在恢复库表 DDL 时添加了 `ifExists` 参数。对于已经存在的库表（视为已经创建完毕），BR 工具会自动跳过恢复。
```

File: br/br-checkpoint-restore.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+日志恢复完成后会重新启动 GC，不需要手动启动。但如果你不想再继续恢复时，请手动开启 GC，开启方法如下：
+
+br 工具暂停 GC 的原理是执行 `SET config tikv gc.ratio-threshold = -1.0`，你可以通过修改 `gc.ratio-threshold` 的值手动开启 GC，例如执行 `SET config tikv gc.ratio-threshold = 1.1` 将参数重置为默认值。
+
+### 部分数据需要重新恢复
+
+在重试恢复时，部分已恢复的数据可能需要重新恢复，包括正在恢复的数据和还未被断点记录的数据。
+
+- 由错误引起的退出，br 工具会在退出前将已恢复的数据元信息持久化到外部存储中，因此只有正在恢复的数据需要在下一次重试中重新恢复。
```
Comment (hfxsd) [inline br/br-checkpoint-restore.md:32]:
```suggestion
- 由错误引起的退出，BR 工具会在退出前将已恢复的数据元信息持久化到外部存储中，因此只有正在恢复的数据需要在下一次重试中重新恢复。
```

File: br/br-checkpoint-restore.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+- 在日志恢复失败后，请避免向集群写入数据、删除或创建表。由于日志备份数据中可能包含重命名表的 DDL 操作，断点恢复无法确认被删除的表或已存在的表是否是外部操作引起的，这会影响下次重试恢复的准确性。
+- 在快照恢复失败后，不推荐对表进行删除操作。
+
+## 实现原理
+
+断点恢复的实现原理分为快照恢复和日志恢复两部分。
+
+### 快照恢复
+
+快照恢复的实现原理与[快照备份](/br/br-checkpoint-backup.md#实现原理)类似，br 工具会批量恢复一段键范围 (Region) 内的所有的 SST 文件。恢复完成后，br 工具记录下这段范围以及该范围对应的恢复集群的表 ID。断点恢复功能会定期将这些新增的相关恢复信息上传至外部存储中，从而持久化存储该恢复任务已完成的键范围。
```
Comment (hfxsd) [inline br/br-checkpoint-restore.md:47]:
```suggestion
快照恢复的实现原理与[快照备份](/br/br-checkpoint-backup.md#实现原理)类似，BR 工具会批量恢复一段键范围 (Region) 内的所有的 SST 文件。恢复完成后，BR 工具记录下这段范围以及该范围对应的恢复集群的表 ID。断点恢复功能会定期将这些新增的相关恢复信息上传至外部存储中，从而持久化存储该恢复任务已完成的键范围。
```

File: br/br-checkpoint-restore.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+在重试恢复时，br 工具会从外部存储读取已经恢复的键范围，并匹配相关的表 ID。在恢复过程中，会跳过与断点恢复记录的键范围重叠且对应表 ID 相同的范围。
```
Comment (hfxsd) [inline br/br-checkpoint-restore.md:49]:
```suggestion
在重试恢复时，BR 工具会从外部存储读取已经恢复的键范围，并匹配相关的表 ID。在恢复过程中，会跳过与断点恢复记录的键范围重叠且对应表 ID 相同的范围。
```

File: releases/release-6.5.0.md [hunk 7]
```diff
@@ -284,7 +284,7 @@ TiDB 6.5.0 为长期支持版本 (Long-Term Support Release, LTS)。
 
     TiDB 快照备份功能支持断点续传。当 BR 遇到可恢复的错误时会进行重试，但是超过固定重试次数之后会备份退出。断点续传功能允许对持续更长时间的可恢复故障进行重试恢复，比如几十分钟的网络故障。
 
-    需要注意的是，如果你没有在 BR 退出后一个小时内完成故障恢复，那么还未备份的快照数据可能会被 GC 机制回收，从而造成备份失败。更多信息，请参考[用户文档](/br/br-checkpoint.md)。
+    需要注意的是，如果你没有在 BR 退出后一个小时内完成故障恢复，那么还未备份的快照数据可能会被 GC 机制回收，从而造成备份失败。更多信息，请参考[用户文档](/br/br-checkpoint-backup.md)。
```
Comment (hfxsd) [inline releases/release-6.5.0.md:287]:
```suggestion
    需要注意的是，如果你没有在 BR 退出后一个小时内完成故障恢复，那么还未备份的快照数据可能会被 GC 机制回收，从而造成备份失败。更多信息，请参考[用户文档](/br/br-checkpoint-backup.md#确保在-GC-前重试)。
```

File: br/br-checkpoint-restore.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+如果在重试恢复前，你删除了某些表，那么重试恢复时新创建的表的 ID 会与之前记录在断点恢复的表 ID 不同，从而绕过使用之前的断点恢复信息，重新恢复该表。
```
Comment (hfxsd) [inline br/br-checkpoint-restore.md:51]: "重新恢复该表" 是否应该改成 “导致无法重新恢复该表”

File: br/br-checkpoint-restore.md (hunk 6 above)
Comment (Oreoxmt) [reply br/br-checkpoint-restore.md:49]: BR 是指备份恢复功能，br 是指 br cli 工具 https://docs.pingcap.com/zh/tidb/stable/glossary#br

File: br/br-checkpoint-restore.md (hunk 8 above)
Comment (Leavrth) [reply br/br-checkpoint-restore.md:51]: `绕过使用之前的断点恢复信息，重新恢复该表`  的意思可以理解为 `无视之前的断点信息，重头恢复该表。` 。`绕过` 一词表示了在这个 ID 匹配的逻辑下，新的 ID 的同个表能够不去使用旧的 ID 的断点恢复信息，而是重新记录新 ID 对应的断点恢复信息。

File: br/br-checkpoint-backup.md [hunk 9]
```diff
@@ -1,17 +1,26 @@
...
+为了尽可能继续上一次的备份，从 TiDB v6.5.0 起，备份恢复特性引入了断点备份的功能。该功能可以在备份意外中断后保留上一次备份的大部分进度。
 
 ## 使用场景
 
-如果你的 TiDB 集群规模很大，无法接受备份失败后需要重新备份的情况，那么，你可以开启断点备份功能。开启该功能后，br 工具会定期记录已备份的分片，使得在下一次重试备份时回到离上一次退出时较近的进度。
+如果你的 TiDB 集群规模很大，无法接受备份失败后需要重新备份的情况，那么，你可以使用断点备份功能重试备份。br 工具会定期记录已备份的分片，使得在下一次重试备份时回到离上一次退出时较近的进度。
+
+## 实现原理
+
+在快照备份过程中，br 工具将表编码成对应的键空间，并生成备份的 RPC 请求发送给所有 TiKV 节点。接收到备份请求后，TiKV 节点会选择请求范围内的数据进行备份。每当 TiKV 节点备份完一个 `region` 级别的数据，就会返回给 br 工具这段范围相关的备份信息。
```
Comment (Oreoxmt) [inline br/br-checkpoint-backup.md:19]:
```suggestion
在快照备份过程中，br 工具将表编码成对应的键空间，并生成备份的 RPC 请求发送给所有 TiKV 节点。接收到备份请求后，TiKV 节点会选择请求范围内的数据进行备份。每当 TiKV 节点备份完一个 Region 级别的数据，就会返回给 br 工具这段范围相关的备份信息。
```

File: releases/release-6.5.0.md [hunk 10]
```diff
@@ -284,7 +284,7 @@ TiDB 6.5.0 为长期支持版本 (Long-Term Support Release, LTS)。
 
     TiDB 快照备份功能支持断点续传。当 BR 遇到可恢复的错误时会进行重试，但是超过固定重试次数之后会备份退出。断点续传功能允许对持续更长时间的可恢复故障进行重试恢复，比如几十分钟的网络故障。
 
-    需要注意的是，如果你没有在 BR 退出后一个小时内完成故障恢复，那么还未备份的快照数据可能会被 GC 机制回收，从而造成备份失败。更多信息，请参考[用户文档](/br/br-checkpoint.md)。
+    需要注意的是，如果你没有在 BR 退出后一个小时内完成故障恢复，那么还未备份的快照数据可能会被 GC 机制回收，从而造成备份失败。更多信息，请参考[用户文档](/br/br-checkpoint-backup.md#确保在-GC-前重试)。
```
Comment (Oreoxmt) [inline releases/release-6.5.0.md:287]:
```suggestion
    需要注意的是，如果你没有在 BR 退出后一个小时内完成故障恢复，那么还未备份的快照数据可能会被 GC 机制回收，从而造成备份失败。更多信息，请参考[用户文档](/br/br-checkpoint-backup.md#确保在-gc-前重试)。
```

File: br/br-checkpoint-restore.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
```
Comment (hfxsd) [inline br/br-checkpoint-restore.md:26]:
```suggestion
如果在重试恢复前，你删除了某些表，那么重试恢复时新创建的表的 ID 会与之前记录在断点恢复的表 ID 不同，从而绕过使用之前的断点恢复信息，重新恢复该表，也就是说，新的 ID 的同个表可以不使用旧的 ID 的断点恢复信息，而是重新记录新 ID 对应的断点恢复信息。
```

**Review by hfxsd — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

File: br/br-checkpoint-restore.md [hunk 12]
```diff
@@ -59,5 +59,4 @@ br 工具暂停 GC 的原理是通过执行 `SET config tikv gc.ratio-threshold
 
 ### 不要在恢复期间修改集群数据
 
-- 在日志恢复失败后，请避免向集群写入数据、删除或创建表。由于日志备份数据中可能包含重命名表的 DDL 操作，断点恢复无法确认被删除的表或已存在的表是否是外部操作引起的，这会影响下次重试恢复的准确性。
-- 在快照恢复失败后，不建议对表进行删除操作。
+在恢复失败后，请避免向集群写入或删除数据、删除或创建表。由于日志备份数据中可能包含重命名表的 DDL 操作，断点恢复无法确认被删除的表或已存在的表是否是外部操作引起的，这会影响下次重试恢复的准确性。
```
Comment (Oreoxmt) [inline br/br-checkpoint-restore.md:62]:
```suggestion
在恢复失败后，请避免向集群写入或删除数据、删除或创建表。由于备份数据中可能包含重命名表的 DDL 操作，断点恢复无法确认被删除的表或已存在的表是否是外部操作引起的，这会影响下次重试恢复的准确性。
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**